src/katgpucbf/xbgpu/correlation.py

```diff
@@ -195,7 +202,10 @@ def __init__(
         )
 
         # TODO: dtypes must depend on input bitwidth
-        self.slots["in_samples"] = accel.IOSlot(dimensions=input_data_dimensions, dtype=np.int8)
+        self.slots["in_samples"] = accel.IOSlot(
+            dimensions=input_data_dimensions,
+            dtype=np.dtype(f"int{self.template._sample_bitwidth}"),
```

Comment:
It's unfortunately not that simple. numpy doesn't have an int4 type; to support int4 you're going to need to restructure things a lot to remove the N_POLS dimension and instead use a dtype that represents a complex value (see katgpucbf.utils.gaussian_dtype). And the 16-bit support in TCC is (if I recall) float16, not int16, with floating-point output. So I don't think there is a lot of value in this change since it's only correct in one case and in that case the original version was correct anyway. Rather just add an `assert sample_bitwidth == 8` and then use np.int8.